### PR TITLE
[ValidatingAdmissionPolicy]Fix flaky test on type checking crd

### PR DESCRIPTION
**Pull request by cici37**
#### What type of PR is this?
/kind flake

#### Which issue(s) this PR fixes:

Fixes #https://github.com/kubernetes/kubernetes/issues/123829

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apimachinery/validatingadmissionpolicy.go [hunk 1]
```diff
@@ -355,7 +355,7 @@ var _ = SIGDescribe("ValidatingAdmissionPolicy [Privileged:ClusterAdmin]", func(
 				if err != nil {
 					return false, err
 				}
-				if policy.Status.TypeChecking != nil {
+				if policy.Status.TypeChecking != nil && len(policy.Status.TypeChecking.ExpressionWarnings) > 0 {
```
Comment (jiahuif) [inline test/e2e/apimachinery/validatingadmissionpolicy.go:358]: No this won't solve the problem. The type checker will not retry when the RESTMapper does not yet recognize the requested CRD. Here, TypeChecking would be empty but non-nil, and the wait will time out.

**Comment by cici37** [comment]
Duplicated with https://github.com/kubernetes/kubernetes/pull/123873

**Pull request closed without merging**